src/repository/index.ts

```diff
@@ -241,12 +249,31 @@ Message: ${err.message}`,
     return obj;
   }
 
+  getFailures(): number {
+    return this.failures;
+  }
+
+  nextFetch(): number {
+    return this.refreshInterval + this.failures * this.refreshInterval;
+  }
+
+  private backoff(): number {
+    this.failures = min(this.failures + 1, 10);
+    return this.nextFetch();
+  }
+
+  private countSuccess(): number {
+    this.failures = max(this.failures - 1, 0);
+    return this.nextFetch();
+  }
+
   async fetch(): Promise<void> {
```

Comment:
`fetch` is now just short of 100 lines. Might be worth breaking this down into smaller methods at this point

Reply:
agreed. Something like what we did for metrics. I'll have a look.